XRTK-Core/Packages/com.xrtk.core/Runtime/Utilities/XRDeviceUtilities.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine.XR;
+
+#if UNITY_2020_1_OR_NEWER
+using UnityEngine;
+using System.Collections.Generic;
+#endif
+
+namespace XRTK.Utilities
+{
+    /// <summary>
+    /// Utilities to determine connected device capabilities and state.
+    /// </summary>
+    public static class XRDeviceUtilities
+    {
+        /// <summary>
+        /// Gets whether an XR display device is currently connected to the machine.
+        /// </summary>
+        public static bool IsDevicePresent
+        {
+            get
+            {
+#if UNITY_2020_1_OR_NEWER
+                var xrDisplaySubsystems = new List<XRDisplaySubsystem>();
```

Comment:
Not sure how I feel about this as we're trying to break away from the XR SDK and its subsystems

Reply:
This should be fine for now, as it's the only realistic way to query Unity if XR is actually running. Practically boilerplate.

It's just replacing the old XR.isPresent subsystem check 

We are definitely NOT using it ,but we do need to know if it's running

Reply:
if it's in the code base there's a chance it'll be used. I'd rather see us try to stay away as much as possible without inadvertently adding a dependency we didn't intent to.

Reply:
These APIs used here are

- native Unity APIs, no package or whatsoever is needed. They are part of the editor installation itself
- The only way to query whether a device is connected
- Might just as well be what XRDevice.isPresent did under the hood for a while now

Unless there is any valid reason not to use it, I don't see why not. It's plain code that simply queries whether a device is connectd. No more, no less.

Reply:
I'm fine with this then, but I do have to warn that this wouldn't work on a platform like webxr, for example, since the xr sdk utterly fails in the webgl builds (something to do with not being able to read the local json configuration files that the xr sdk utilizes to setup the subsystems)

Reply:
I'd much rather see our own version of this API implemented

Reply:
Sure, get a job at Unity then @StephenHodgson and open up their engine to allow us to enable and test for XR, simples...
P.S> I'd also like an RX3090, getting Unity to open up would certainly be easier.

Reply:
Believe me I've tried. They said they had all the help they needed 😅